zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/clock/ClockProcessor.java

```diff
@@ -50,6 +54,16 @@ public void processNewCommand(final TypedRecord<ClockRecord> command) {
     final var intent = (ClockIntent) command.getIntent();
     final var resultIntent = followUpIntent(intent);
 
+    if (intent == ClockIntent.PIN && clockRecord.getTime() < 0) {
```

Comment:
🔧 You can move this validation check some lines above, you need only `clockRecord` and `intent` as variable, you can define the others variable after this check

Reply:
Yeah, moved validation upper